### ISSUE TITLE
Istio-Ingressgateway fails during startup when trying to configure certain global mesh values

### ISSUE THREAD

**Issue by javadebo** [istio/istio#27464]
When trying to set global mesh values like:
ingressService: istio-ingressgateway
ingressControllerMode: STRICT
ingressClass: external-istio

The istio-ingressgateway fails during startup. The pods puts themselves in a "completed" state, and then restarts themselves over and over again. 

The configuration has been tested in different areas of the configuration. Both under meshConfig in IstioOperatorSpec, and under values, and under values.global, nothing works. This was in previous releases possible to configure. 

[X ] Docs
[X ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

It should be possible to configure these values, since in some use cases we need to use many instances of istio. And in this case, some are NOT configured to be STRICT, and therefor reads all Ingress objects. We need a way to set a specific Istio to listen to a specific Ingress object(with annotations)

Add this to the IstioOperator CRD.

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istio-control-plane
spec:
#  meshConfig:
#    ingressClass: external-istio
  components:
    base:
      enabled: true
    pilot:
      enabled: true
    ingressGateways:
      - name: istio-ingressgateway
        enabled: true
        k8s:
          service:
            type: LoadBalancer
            loadBalancerIP: 185.16.192.58
            ports:
              - port: 15020
                name: status-port
              - port: 80
                name: http2
              - port: 443
                name: https
          replicaCount: 2
  addonComponents:
    kiali:
      enabled: true
    grafana:
      enabled: false
    prometheus:
      enabled: true
    tracing:
      enabled: false
    istiocoredns:
      enabled: false
  values:
    #  meshConfig:
    #    ingressClass: external-istio
    global:
#      ingress:
#        ingressService: istio-ingressgateway
#        ingressControllerMode: STRICT
#        ingressClass: external-istio
      logging:
        level: default:info
    #      k8sIngress:
    #        enabled: false
    #        gatewayName: ingressgateway
    #        enableHttps: true
    pilot:
#      ingress:
#        ingressService: istio-ingressgateway
#        ingressControllerMode: STRICT
#        ingressClass: external-istio
```

Uncomment any of these, and no one will work. According to documentation(which is very very poorly written), it should be possible to set this values either by overriding values.yaml, or in meshConfig, nothing works. 

client version: 1.7.1
control plane version: 1.7.1
data plane version: 1.7.1 (1 proxies)

Istio was installed using IstioOperator 1.7.1

Google Cloud Kubernetes Engine

**Comment by shamsher31** [comment]
These are still available in `1.7` under `meshConfig`
https://github.com/istio/istio/blob/a8c04f92e236676977b3ed58132437d9fafc9aed/pkg/config/mesh/mesh.go#L97-L99

I have slightly modified the minimal profile to test it.

```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    ingressService: istio-ingressgateway
    ingressControllerMode: STRICT
    ingressClass: external-istio
  components:
    ingressGateways:
    - name: istio-ingressgateway
      enabled: true
```
```
$ ./out/linux_amd64/istioctl install --set profile=minimal --set hub=gcr.io/istio-testing -d manifests/
✔ Istio core installed                                                                                         
✔ Istiod installed                                                                                             
✔ Ingress gateways installed                                                                                   
✔ Installation complete
```

```
$ kubectl get pods -n istio-system
NAME                                    READY   STATUS    RESTARTS   AGE
istio-ingressgateway-74b98f8d9c-4tqzv   1/1     Running   0          13s
istiod-6fc8557fbc-hvq5d                 1/1     Running   0          17s
```
And I can see the ingress is configured.
```
$ kubectl logs istio-ingressgateway-74b98f8d9c-4tqzv -n istio-system

enablePrometheusMerge: true

ingressClass: external-istio
ingressControllerMode: STRICT
ingressService: istio-ingressgateway

rootNamespace: istio-system
trustDomain: cluster.local
```

```
$ kubectl logs istiod-6fc8557fbc-5n4b2 -n istio-system

2020-09-29T07:04:22.061869Z     info    mesh configuration: {
    "proxyListenPort": 15001,
    "connectTimeout": "10s",
    "protocolDetectionTimeout": "5s",

    "ingressClass": "external-istio",
    "ingressService": "istio-ingressgateway",
    "ingressControllerMode": "STRICT",

    "enableTracing": true
}
```

@javadebo can you provide your logs?

**Comment by howardjohn** [comment]
@javadebo we cannot fix this without the logs from the failure, are you able to provide them?
